concordium-sdk/src/main/java/com/concordium/sdk/transactions/CredentialRegistrationId.java

```diff
@@ -0,0 +1,49 @@
+package com.concordium.sdk.transactions;
+
+import lombok.*;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+
+import java.util.Objects;
+
+/**
+ * A credential registration id
+ */
+@ToString
+@EqualsAndHashCode
+public class CredentialRegistrationId {
+    //the length of `CredentialRegistrationId` bytes
+    private static final int LENGTH = 48;
+    @Getter
+    private final byte[] regId;
+
+    private CredentialRegistrationId(byte[] regId) {
+        this.regId = regId;
+    }
+
+    /**
+     * Create a {@link CredentialRegistrationId} from the hex encoded registration id.
+     */
+    public static CredentialRegistrationId from(String encoded) {
+        try {
+            val regId = Hex.decodeHex(encoded);
```

Comment:
This is an API question, does this fail in cases like
```
aabb$
```
i.e., where the start is correct, but there is some leftover?

Reply:
Yes. The Hex.decodeHex(hex) will throw a `DecoderException("Odd number of characters.")` which will be propagated via the `IllegalArgumentException` being thrown below.